src/v/net/server.cc

```diff
@@ -137,9 +156,19 @@ ss::future<> server::accept(listener& s) {
                     SOL_SOCKET, SO_SNDBUF, &send_buf, sizeof(send_buf));
               }
 
+              auto name = s.name;
+              if (_connection_rates) {
+                  try {
+                      co_await _connection_rates->maybe_wait(
+                        ar.remote_address.addr());
+                  } catch (const std::exception& e) {
+                      co_return ss::stop_iteration::no;
```

Comment:
Would be good to emit a log message and increment a prometheus counter when we do this.